src/Model/GroundWaterFlow/gwf3ic8.f90

```diff
@@ -74,175 +69,77 @@ subroutine ic_ar(this, x)
     real(DP), dimension(:), intent(inout) :: x
     ! -- locals
     integer(I4B) :: n
-! ------------------------------------------------------------------------------
     !
-    ! -- Print a message identifying the initial conditions package.
-    write (this%iout, 1) this%inunit
-1   format(1x, /1x, 'IC -- INITIAL CONDITIONS PACKAGE, VERSION 8, 3/28/2015', &
-           ' INPUT READ FROM UNIT ', i0)
-    !
-    ! -- Allocate arrays
+    ! -- allocate arrays
     call this%allocate_arrays(this%dis%nodes)
     !
-    ! -- Read options
-    call this%read_options()
-    !
-    ! -- Read data
-    call this%read_data()
+    ! -- read grid data
+    call this%source_griddata()
     !
-    ! -- Assign x equal to strt
+    ! -- assign starting head
```

Comment:
Maybe we couldn't do this in ic_cr, which is why we are calling ic_load() as part of the AR calls.  Maybe we can keep an ic_ar() routine and assign strt to model%x in it.  I know it is not a "allocate and read" action, but maybe we keep the names consistent for now and consider renaming them later.

Reply:
cf https://github.com/MODFLOW-USGS/modflow6/pull/1425#discussion_r1382401652